### PR TITLE
DT-1249 Improve Swagger documentation. Split out private APIs and add…

### PR DESCRIPTION
**Pull request by andymarke**
The changes are:

1. Split out the OAuth2 protected documents from the private APIs used by NewTech into different document sets
2. Added documentation about the main base endpoints fir each environments
3. Cleaned up some of the section names - will do more in future PR
4. Added default messages for 401/403 responses
4. Upgrade to springfox 3

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/config/SwaggerConfig.java [hunk 1]
```diff
@@ -32,18 +33,42 @@
...
+                .paths(
+                        regex("(\\/ping.*)")
+                                .or(regex("(\\/info.*)"))
+                                .or(regex("(\\/health.*)"))
+                                .or(regex("(\\/secure/.*)")))
+                .build()
+                .genericModelSubstitutes(Optional.class)
+                .directModelSubstitute(ZonedDateTime.class, java.util.Date.class)
+                .directModelSubstitute(LocalDateTime.class, java.util.Date.class)
+                .globalResponses(HttpMethod.GET, getCustomizedResponseMessages())
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/config/SwaggerConfig.java:52]: Add some defaults for 401 and 403 errors which are the same across the service

File: src/main/java/uk/gov/justice/digital/delius/config/SwaggerConfig.java [hunk 2]
```diff
@@ -32,18 +33,42 @@
...
+                .useDefaultResponseMessages(false)
+                .groupName("NewTech Private APIs")
+                .apiInfo(newTechApiInfo())
+                .select()
+                .apis(RequestHandlerSelectors.any())
+                .paths(
+                        regex("(\\/ping.*)")
+                                .or(regex("(\\/info.*)"))
+                                .or(regex("(\\/health.*)"))
+                                .or(regex("(\\/api/.*)")))
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/config/SwaggerConfig.java:71]: private APIs used by NewTech

File: src/main/java/uk/gov/justice/digital/delius/config/SwaggerConfig.java [hunk 3]
```diff
@@ -61,11 +86,43 @@ private Contact contactInfo() {
...
+                        "<p>There is cross-over with the <b>prison-api</b> though suspects on remand will not be surfaced by this API unless that have previously been on probation.</p>" +
+                        "<div>" +
+                        "This service is secured by <b>OAuth2</b> with tokens supplied by HMPPS Auth. Most read-only endpoints require the <b>ROLE_COMMUNITY</b> to access, but check each endpoint where this differs." +
+                        "<p>This service can be accessed in a number environments. For each environment a different set of OAuth2 credentials from HMPPS Auth are required</p>" +
+                        "<ul>" +
+                        "<li>Test/Development: <b>https://community-api-secure.test.delius.probation.hmpps.dsd.io</b></li>" +
+                        "<li>Pre-production: <b>https://community-api-secure.pre-prod.delius.probation.hmpps.dsd.io</b></li>" +
+                        "<li>Production: <b>https://community-api-secure.probation.service.justice.gov.uk</b></li>" +
+                        "</ul>" +
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/config/SwaggerConfig.java:99]: Couldn't see another way of specifying this URLs, there is even a new` Docklet.servers()` method  that has been added but to doesn't produce any swagger

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/AttendanceResource.java [hunk 4]
```diff
@@ -48,8 +48,6 @@ public AttendanceResource (final AttendanceService attendanceService, final Offe
     @ApiResponses(
             value = {
                     @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
-                    @ApiResponse(code = 401, message = "Unauthorised", response = ErrorResponse.class),
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/AttendanceResource.java:51]: removed 401/403 just to prove the global ones work (which it does)

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java [hunk 5]
```diff
@@ -483,11 +483,11 @@ public Nsi getNsiByNsiId(
 
     @ApiOperation(value = "Return pageable list of all offender identifiers that match the supplied filter")
     @ApiImplicitParams({
-            @ApiImplicitParam(name = "page", dataType = "int", paramType = "query",
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java:486]: Springfox 3 upgrade validation fix

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**